packages/application-generic/src/services/cache/key-builders/entities.ts

```diff
@@ -80,7 +81,7 @@ const buildGroupedBlueprintsKey = (): string =>
     identifier: BLUEPRINT_IDENTIFIER,
   });
 
-const buildAuthServiceKey = ({ apiKey }: { apiKey: string }): string =>
+const buildAuthServiceKey = ({ apiKey }: { apiKey: EncryptedSecret }): string =>
```

Comment:
the ticket says to store the hash in the cache, I guess just to save memory?

Reply:
True please note the comment i left https://github.com/novuhq/novu/pull/5046#issuecomment-1875520097 regarding the ticket. i am not sure if we will benefit too much from a hash because the key is relatively small, but I am not open to suggestions.

Reply:
```
apiKey nvsk.41e558c55f4c1d18f45354bbcfc31c87:7db2803321f4d5b6db48625505621d922dda7a09623361ca2947f13d057ce586d2666bbfb5afabb30613be5743681239
apiKey - string size 134

apiKeyHash 9c2e166d6bc2f95fb8f92442c22e54edb3cd0529c4260af3c9f4c5f3fe98c6cb
apiKeyHash - string size 64
```

it wouldn't be a huge memory win for us (~4mb if all envs cached) but still, it's 2 times smaller in size...

[codesandbox link](https://codesandbox.io/p/devbox/encryption-9xvr3w?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clqyzv07l00063b6jfgu0uvl1%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clqyzv07k00023b6jxr53oi7g%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clqyzv07l00043b6jfgu7omra%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clqyzv07l00053b6j5prsb177%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clqyzv07k00023b6jxr53oi7g%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqyzv07k00013b6jsd57vg3s%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%257D%255D%252C%2522id%2522%253A%2522clqyzv07k00023b6jxr53oi7g%2522%252C%2522activeTabId%2522%253A%2522clqyzv07k00013b6jsd57vg3s%2522%257D%252C%2522clqyzv07l00053b6j5prsb177%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clqyzv07l00053b6j5prsb177%2522%257D%252C%2522clqyzv07l00043b6jfgu7omra%2522%253A%257B%2522id%2522%253A%2522clqyzv07l00043b6jfgu7omra%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqyzv07k00033b6jzgsuhabw%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clqyzv07k00033b6jzgsuhabw%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

Reply:
omg i hope it did not take too much time to create the sandbox, well why not 😊,  i will look into it. 
Thanks for the comparison help!